### ISSUE TITLE
Rename {N} Observable to avoid confusion with standard rxjs Observable.

### ISSUE THREAD

**Issue by NathanWalker** [NativeScript/NativeScript#3214]
[RxJS](https://github.com/ReactiveX/rxjs) is *not just* used with Angular implementations. It is used across many different types of JavaScript projects, has become a standard, is rich in features, and is familiar to many.

The {N} [Observable](https://docs.nativescript.org/api-reference/classes/_data_observable_.observable.html) class causes much confusion to {N} newcomers because of this.

Please upvote this issue if you agree: https://nativescript.ideas.aha.io/ideas/NS-I-154

### My suggestion on rename (open to many opinions, please share!)

The only place I've encountered them is when dealing with view bindings/view model files (there may be other cases). But in the event they are primarily focused on view model implementations, I would take into consideration the community has largely adopted a prefix for NativeScript specific classes, therefore my vote for rename:

`TNSEventSource`

Then view models would be updated to:

```
export class ViewModel extends TNSEventSource {
  // could still use this.set('property', value);
}
```

However, we would all love to get away from the `this.set('property', value);` necessity.

@hdeshev, could you think of a way that `TNSEventSource` (or some other name replacement class) could auto detect property changes from `viewModelInstance.property = value` removing the need for the `set` semantics?

**Comment by NathanWalker** [comment]
Other Ref: https://github.com/NativeScript/NativeScript/issues/1289

**Comment by NathanaelA** [comment]
To be honest; I would think it would be better to see how easily it would be to actually replace the observable code base with RxJS so there is only one event/observable system in {N} and/or Angular {N}.

In addition to answer your last question @NathanWalker, you can use the new built in ES6 stuff like RxJS does to observe changes and proxy changes to things, and then eliminate any manual sets...   Both device engines I believe support ES6 almost fully.

**Comment by NathanWalker** [comment]
I agree with @NathanaelA I feel like an effort to swap *all* observable and event stream code with standard RxJS would be worth the effort. It would definitely be some work but well worth it.

**Comment by hdeshev** [comment]
I'm all for renaming {N} Observable to `TNSEventSource`. I'm not sure about this replace thing though. Do we have to change all View objects (those are observable) into RxJS observables? How will the API look like?


---
Thinking out lout here...

There's also the problem of program size. RxJS is a **fat** library, and in order to use it effectively, we'd need to have everybody use webpack in their project. I'm not sure we are ready for that yet.

We also need performance tests for change events and binding updates...

**Comment by NathanaelA** [comment]
@hdeshev - Agreed I think a lot of benchmarks need to be done.  This might be a non-starter depending on the benchmarks!   If RXJS adds any real overhead, then I would only encourage the renaming.   ;-)   

 My idea is basically it subsumes the behavior so that the rest of the code base really doesn't change; even if we have to add a couple minor monkey patches to RXJS to make it drop in compatible.   Technically you create a new data/observable.js & OArray.js file that requires RXJS and exposes RXJS's Observable as the normal Observable.    So the rest of the code base just works, but we get the full power of RXJS.

As for size; you are correct, it adds another meg of size; so I would agree WebPack would be important, but isn't WebPack supposed to be a high priority already?

**Comment by NathanWalker** [comment]
@hdeshev Curious to revitalize this discussion - In regards to swapping data/Observable with Rxjs - sizable and enormous task? Yes. Worth it. I'm almost certain I'm not alone in thinking it would be well worth it.

**Comment by NickIliev** [comment]
^^^ @vakrilov @manoldonev @MartoYankov @ADjenkov @sis0k0 @vchimev 

@NathanWalker just FYI: @hdeshev is not part of the NativeScript team.

**Comment by larssn** [comment]
RxJS is certainly modular now (since v6), so size shouldn't be an issue; even without tree shaking.

**Comment by insytes** [comment]
👀
